xbmc/video/VideoInfoScanner.cpp

```diff
@@ -237,201 +246,159 @@ namespace VIDEO
-        items.SetPath(strDirectory);
-        GetPathHash(items, hash);
-        bSkip = true;
-        if (!m_database.GetPathHash(strDirectory, dbHash) || !StringUtils::EqualsNoCase(dbHash, hash))
-          bSkip = false;
-        else
-          items.Clear();
-      }
-      else
-      {
-        CFileItemPtr item(new CFileItem(URIUtils::GetFileName(strDirectory)));
-        item->SetPath(strDirectory);
-        item->m_bIsFolder = true;
-        items.Add(item);
-        items.SetPath(URIUtils::GetParentPath(item->GetPath()));
-      }
-    }
-    bool foundSomething = false;
-    if (!bSkip)
+  bool CVideoInfoScanner::DoScan(const std::string& strDirectory){
+    return false;
+  }
+  bool CVideoInfoScanner::DoScan(const std::vector<SourcePathContent>& _pathsContent)
+  {
+    if (m_handle)
     {
-      foundSomething = RetrieveVideoInfo(items, settings.parent_name_root, content);
-      if (foundSomething)
-      {
-        if (!m_bStop && (content == CONTENT_MOVIES || content == CONTENT_MUSICVIDEOS))
-        {
-          m_database.SetPathHash(strDirectory, hash);
-          if (m_bClean)
-            m_pathsToClean.insert(m_database.GetPathId(strDirectory));
-          CLog::Log(LOGDEBUG, "VideoInfoScanner: Finished adding information from dir {}",
-                    CURL::GetRedacted(strDirectory));
-        }
-      }
-      else
-      {
-        if (m_bClean)
-          m_pathsToClean.insert(m_database.GetPathId(strDirectory));
-        CLog::Log(LOGDEBUG, "VideoInfoScanner: No (new) information was found in dir {}",
-                  CURL::GetRedacted(strDirectory));
-      }
-    }
-    else if (!StringUtils::EqualsNoCase(hash, dbHash) && (content == CONTENT_MOVIES || content == CONTENT_MUSICVIDEOS))
-    { // update the hash either way - we may have changed the hash to a fast version
-      m_database.SetPathHash(strDirectory, hash);
+      m_handle->SetTitle(g_localizeStrings.Get(20415));  //"Scanning for new content"
     }
 
-    if (m_handle)
-      OnDirectoryScanned(strDirectory);
 
-    for (int i = 0; i < items.Size(); ++i)
-    {
-      CFileItemPtr pItem = items[i];
+    // Calculate total items count to display a proper progress bar
+    int totalItems = 0;
+    for (const auto& pathContent : _pathsContent) {
+      totalItems += pathContent.items.Size();
+    }
 
-      if (m_bStop)
-        break;
+    int itemsChecked = 0;
+    for (const auto& pathContent : _pathsContent) {
+      const CFileItemList& remoteDirectoryItems = pathContent.items;
 
-      // add video extras to library
-      if (foundSomething && !m_ignoreVideoExtras && pItem->IsVideoExtras())
+      for (int i = 0; i < remoteDirectoryItems.Size(); ++i)
       {
-        if (AddVideoExtras(items, content, pItem->GetPath()))
-        {
-          CLog::Log(LOGDEBUG, "VideoInfoScanner: Finished adding video extras from dir {}",
-                    CURL::GetRedacted(pItem->GetPath()));
+        CFileItemPtr remoteItem = remoteDirectoryItems[i];
```

Comment:
Please use the real type `std::shared_ptr<CFileItem>`. We want to get rid of the `CFileItemPtr` typedef.